packages/typegpu/src/core/declare/tgpuDeclare.ts

```diff
@@ -0,0 +1,73 @@
+import type { TgpuNamable } from '../../namable';
+import type { ResolutionCtx, TgpuResolvable } from '../../types';
+import {
+  type ExternalMap,
+  applyExternals,
+  replaceExternalsInWgsl,
+} from '../resolve/externals';
+
+// ----------
+// Public API
+// ----------
+
+/**
+ * Extra declaration that shall be included in final WGSL code,
+ * when resolving objects that use it.
+ */
+export interface TgpuDeclare extends TgpuResolvable, TgpuNamable {
```

Comment:
Tiny nit, does not have to be namable. If we find that its helpful for debugging purposes, we can bring it back.
```suggestion
export interface TgpuDeclare extends TgpuResolvable {
```